packages/contracts-bedrock/scripts/deploy/DeployProxy.s.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Forge
+import { Script } from "forge-std/Script.sol";
+
+// Scripts
+import { BaseDeployIO } from "scripts/deploy/BaseDeployIO.sol";
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+
+// Interfaces
+import { IProxy } from "src/universal/interfaces/IProxy.sol";
+
+/// @title DeployProxyInput
+contract DeployProxyInput is BaseDeployIO {
+    // Specify the owner of the proxy that is being deployed
+    address internal _owner;
+
+    function set(bytes4 _sel, address _value) public {
+        if (_sel == this.owner.selector) {
+            require(_value != address(0), "DeployProxy: owner cannot be empty");
+            _owner = _value;
+        } else {
+            revert("DeployProxy: unknown selector");
+        }
+    }
+
+    function owner() public view returns (address) {
+        require(_owner != address(0), "DeployProxy: owner not set");
+        return _owner;
+    }
+}
+
+/// @title DeployProxyOutput
+contract DeployProxyOutput is BaseDeployIO {
+    IProxy internal _proxy;
+
+    function set(bytes4 _sel, address _value) public {
+        if (_sel == this.proxy.selector) {
+            require(_value != address(0), "DeployProxy: proxy cannot be zero address");
+            _proxy = IProxy(payable(_value));
+        } else {
+            revert("DeployProxy: unknown selector");
+        }
+    }
+
+    function checkOutput(DeployProxyInput _mi) public {
+        DeployUtils.assertValidContractAddress(address(_proxy));
+        assertValidDeploy(_mi);
+    }
+
+    function proxy() public view returns (IProxy) {
+        DeployUtils.assertValidContractAddress(address(_proxy));
+        return _proxy;
+    }
+
+    function assertValidDeploy(DeployProxyInput _mi) public {
+        assertValidProxy(_mi);
+    }
+
+    function assertValidProxy(DeployProxyInput _mi) internal {
+        IProxy prox = proxy();
+        vm.prank(_mi.owner());
```

Comment:
q for op-deployer folks - can I use cheatcodes in these scripts? I remember something about most of this living on chain at some point

Reply:
You can use cheatcodes in scripts! You just can't use them in the deployed contracts.

Reply:
For sure - I thought I remembered that at some point we wanted to make these modules live on chain. Disregard, ty!